### PR TITLE
build: bump deps

### PR DESCRIPTION
**Pull request by amaanq**

### CODE REVIEW

**Comment by clason** [comment]
No wasmtime bump? (Current is 25.0.1, which Neovim already tracks.)

File: Cargo.toml [hunk 1]
```diff
@@ -34,15 +34,20 @@ codegen-units = 1    # Maximum size reduction optimizations.
...
-inherits = "optimize"
-strip = false
+[profile.release-dev]
+inherits = "release"
+lto = false
+debug = true
+debug-assertions = true
+overflow-checks = true
+incremental = true
+codegen-units = 256
```
Comment (clason) [inline Cargo.toml:44]: Sneaky manual change!

Comment (amaanq) [reply Cargo.toml:44]: (it's very useful for debugging and profiling, more so than the old one)

Comment (clason) [reply Cargo.toml:44]: Sure, no objection to the change itself; I'm just pointing out that it's hidden in an unrelated PR (which is bad for discoverability and bisecting).

**Pull request merged**